File: javascript/bob/bob.js
var Bob = function() {
  function isBlank(string) {
    return /^\s*$/.test(string);
  }

  function isUpperCase(string) {
    if (!hasLetters(string)) return false;
    return string === string.toUpperCase() ? true : false; 
  }

  function isQuestion(string) {
    return string[string.length - 1] === '?' ? true : false;
  }

  function hasLetters(string) {
    return /[a-zA-Z]/.test(string)
  }

  return {
    hey: function(input) {
      if (isBlank(input)) {
        return "Fine. Be that way!";
      } else if (isUpperCase(input)) {
        return "Whoa, chill out!";
      } else if (isQuestion(input)) {
        return "Sure."
      } else {
        return "Whatever.";
      }
    }
  }
}

module.exports = Bob;

File: python/bob/bob.py
import re

def hey(what):
    if is_question(what):
        return 'Sure.'
    elif is_yelling(what):
        return 'Whoa, chill out!'
    elif is_empty(what):
        return 'Fine. Be that way!'
    else:
        return 'Whatever.'

def is_question(what):
    return re.search(r'\w*\?$', what) and not is_yelling(what)

def is_yelling(what):
    return re.search(r'[a-zA-Z]+', what) and what.upper() == what and not is_empty(what)

def is_empty(what):
    return re.search(r'^\s*$', what)
    